app/controllers/cart_controller.rb

```diff
@@ -22,19 +22,34 @@ def remove_item
     redirect_to '/cart'
   end
 
+  def edit_quantity
+    adjust_quantity
+  end
+
   private
 
   def require_not_admin
       render file: "/public/404" if current_admin
   end
 
-  # def increment_decrement
-  #   if params[:increment_decrement] == "increment"
-  #     cart.add_quantity(params[:item_id]) unless cart.limit_reached?(params[:item_id])
-  #   elsif params[:increment_decrement] == "decrement"
-  #     cart.subtract_quantity(params[:item_id])
-  #     return remove_item if cart.quantity_zero?(params[:item_id])
-  #   end
-  #   redirect_to "/cart"
-  # end
+  def item
+    @item = Item.find(params[:item_id])
+  end
+
+  def adjust_quantity
+    if params[:quantity] == "add" && item.inventory > session[:cart][params[:item_id]]
+      session[:cart][params[:item_id]] += 1
+      redirect_to '/cart'
+    elsif params[:quantity] == "add"
+      flash[:error] = "Out of Stock"
+      redirect_to '/cart'
+    elsif params[:quantity] == "subtract" && session[:cart][params[:item_id]] > 1
+      session[:cart][params[:item_id]] -= 1
+      redirect_to '/cart'
+    else params[:quantity] == "subtract" && session[:cart][params[:item_id]] == 1
```

Comment:
Quantity equal to one, then remove item from cart using helper method from above in the controller.